src/app/clusters/access-control-server/access-control-server.cpp

```diff
@@ -428,6 +467,99 @@ void AccessControlAttribute::OnEntryChanged(const SubjectDescriptor * subjectDes
     ChipLogError(DataManagement, "AccessControlCluster: event failed %" CHIP_ERROR_FORMAT, err.Format());
 }
 
+#if CHIP_CONFIG_USE_ACCESS_RESTRICTIONS
+CHIP_ERROR AccessControlAttribute::ReadCommissioningArl(AttributeValueEncoder & aEncoder)
+{
+    auto accessRestrictionProvider = GetAccessControl().GetAccessRestrictionProvider();
+    if (accessRestrictionProvider == nullptr)
+    {
+        return CHIP_ERROR_NOT_IMPLEMENTED;
+    }
+
+    return aEncoder.EncodeList([&](const auto & encoder) -> CHIP_ERROR {
+        auto entries = accessRestrictionProvider->GetCommissioningEntries();
+
+        for (auto & entry : entries)
+        {
+            ArlEncoder::CommissioningEncodableEntry encodableEntry(entry);
+            ReturnErrorOnFailure(encoder.Encode(encodableEntry));
+        }
+        return CHIP_NO_ERROR;
+    });
+}
+
+CHIP_ERROR AccessControlAttribute::ReadArl(AttributeValueEncoder & aEncoder)
+{
+    auto accessRestrictionProvider = GetAccessControl().GetAccessRestrictionProvider();
+    if (accessRestrictionProvider == nullptr)
+    {
+        return CHIP_ERROR_NOT_IMPLEMENTED;
+    }
+
+    return aEncoder.EncodeList([&](const auto & encoder) -> CHIP_ERROR {
+        for (auto & info : Server::GetInstance().GetFabricTable())
+        {
+            auto fabric = info.GetFabricIndex();
+            // get entries for fabric
+            std::vector<AccessRestrictionProvider::Entry> entries;
+            ReturnErrorOnFailure(accessRestrictionProvider->GetEntries(fabric, entries));
+            for (auto & entry : entries)
+            {
+                ArlEncoder::EncodableEntry encodableEntry(entry);
+                ReturnErrorOnFailure(encoder.Encode(encodableEntry));
+            }
+        }
+        return CHIP_NO_ERROR;
+    });
+}
+void AccessControlAttribute::CommissioningRestrictionListChanged()
+{
+    MatterReportingAttributeChangeCallback(0, AccessControlCluster::Id, AccessControlCluster::Attributes::CommissioningARL::Id);
+}
+
+void AccessControlAttribute::RestrictionListChanged(FabricIndex fabricIndex)
+{
+    CHIP_ERROR err;
+
+    MatterReportingAttributeChangeCallback(0, AccessControlCluster::Id, AccessControlCluster::Attributes::Arl::Id);
+
+    ArlChangedEvent event{ .fabricIndex = fabricIndex };
+
+    EventNumber eventNumber;
+    SuccessOrExit(err = LogEvent(event, 0, eventNumber));
+
+    return;
+
+exit:
+    ChipLogError(DataManagement, "AccessControlCluster: restriction event failed %" CHIP_ERROR_FORMAT, err.Format());
+}
+
+void AccessControlAttribute::OnFabricRestrictionReviewUpdate(FabricIndex fabricIndex, uint64_t token, const char * instruction,
+                                                             const char * redirectUrl)
+{
+    CHIP_ERROR err;
+    ArlReviewEvent event{ .token = token, .fabricIndex = fabricIndex };
+
+    if (instruction != nullptr)
+    {
+        event.instruction.SetNonNull(chip::CharSpan(instruction, strnlen(instruction, kMaxInstructionStringLength)));
```

Comment:
```suggestion
        event.instruction.SetNonNull(CharSpan(instruction, strnlen(instruction, kMaxInstructionStringLength)));
```
This was in the previous set of review comemnts....

Reply:
done